### PR TITLE
[Merged by Bors] - perf: replace many instances of 'linarith' with 'omega'

### PR DESCRIPTION
**Pull request by dwrensha**
Replaces 63 usages of `linarith` or `nlinarith` with `omega`. In all of these cases I have observed the new proof to be faster.

I found these improvements by running [tryAtEachStep](https://github.com/dwrensha/tryAtEachStep) to try `omega` at every tactic step in mathlib.

This is a continuation of the work from #11093.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dwrensha** [comment]
!bench

File: Mathlib/Algebra/Lie/Weights/RootSystem.lean [hunk 1]
```diff
@@ -119,7 +119,7 @@ lemma rootSpace_neg_nsmul_add_chainTop_of_lt (hα : α.IsNonZero) {n : ℕ} (hn
       neg_add_rev, apply_coroot_eq_cast' α β, Int.cast_sub, Int.cast_mul, Int.cast_ofNat,
       mul_comm (2 : K), add_sub_cancel, neg_neg, add_sub, Nat.cast_inj,
       eq_sub_iff_add_eq, ← Nat.cast_add, ← sub_eq_neg_add, sub_eq_iff_eq_add] at this
-    linarith [this, hn]
+    omega
```
Comment (grunweg) [inline Mathlib/Algebra/Lie/Weights/RootSystem.lean:122]: This change seems to be within noise, but it's shorter - I'll take it.

File: Mathlib/GroupTheory/Coxeter/Length.lean [hunk 2]
```diff
@@ -299,36 +299,36 @@ theorem isLeftDescent_iff {w : W} {i : B} :
...
 theorem not_isRightDescent_iff {w : W} {i : B} :
     ¬cs.IsRightDescent w i ↔ ℓ (w * s i) = ℓ w + 1 := by
   unfold IsRightDescent
   constructor
   · intro _
-    exact (cs.length_mul_simple w i).resolve_right (by linarith)
+    exact (cs.length_mul_simple w i).resolve_right (by omega)
   · intro _
-    linarith
+    omega
```
Comment (grunweg) [inline Mathlib/GroupTheory/Coxeter/Length.lean:331]: This file even sped up by 25%, nice!

File: Mathlib/RingTheory/RingHom/Integral.lean [hunk 3]
```diff
@@ -57,7 +57,7 @@ theorem isIntegral_ofLocalizationSpan :
     IsLocalization.map_eq_zero_iff (.powers (f t))] at hp'
   obtain ⟨⟨x, m, (rfl : algebraMap R S t ^ m = x)⟩, e⟩ := hp'
   by_cases hp' : 1 ≤ p.natDegree; swap
-  · obtain rfl : p = 1 := eq_one_of_monic_natDegree_zero hp (by nlinarith)
+  · obtain rfl : p = 1 := eq_one_of_monic_natDegree_zero hp (by omega)
```
Comment (grunweg) [inline Mathlib/RingTheory/RingHom/Integral.lean:60]: This change speeds up the file by 8%, nice!

**Review by grunweg — COMMENTED** [review]
Thanks a lot for this change, would review again! I double-check against the benchmarking output that all changed files really got faster (some very substantially so).
If you find more of these, I'll be happy to review that PR also.
maintainer merge

File: Mathlib/Algebra/Lie/Weights/RootSystem.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Algebra/Lie/Weights/RootSystem.lean:122]: Actually: perhaps this change didn't get benchmarked yet --- that would explain things.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by dwrensha** [comment]
I maybe should have also mentioned:
`tryAtEachStep` reported that in all of these cases the new proof term is smaller than the old one (based on length of pretty-printed string).

`tryAtEachStep` does not report elaboration time or kernel checking time, but that's something I'd like to add.

**Pull request closed without merging**